cpp/src/fil/fil.cu

```diff
@@ -636,6 +637,89 @@ int tree2fil_sparse(std::vector<fil_node_t>& nodes, int root,
   return root;
 }
 
+struct level_entry {
+  int n_branches, n_leaves;
```

Comment:
I'm not sure if "branch" is a standard term for an inner node. I think `n_inner_nodes` would be clearer.

Reply:
made it `n_branch_nodes` in my new patch